code/game/objects/items/inducer.dm

```diff
@@ -7,212 +7,210 @@
 	var/recharging = FALSE
 
 /obj/item/inducer/Initialize(mapload)
 	. = ..()
-	if(!powerdevice && cell_type)
-		powerdevice = new cell_type
 
-/obj/item/inducer/proc/induce(obj/item/stock_parts/power_store/target, coefficient)
-	var/obj/item/stock_parts/power_store/our_cell = get_cell()
-	var/rating_base = target.rating_base
-	var/totransfer = min(our_cell.charge, (rating_base * coefficient * power_transfer_multiplier))
-	var/transferred = target.give(totransfer)
+	if(ispath(powerdevice))
+		powerdevice = new powerdevice(src)
 
-	our_cell.use(transferred)
-	our_cell.update_appearance()
-	target.update_appearance()
+	register_context()
 
-/obj/item/inducer/get_cell()
-	return powerdevice
+	update_appearance(UPDATE_OVERLAYS)
 
-/obj/item/inducer/emp_act(severity)
+/obj/item/inducer/Destroy(force)
+	QDEL_NULL(powerdevice)
 	. = ..()
-	var/obj/item/stock_parts/power_store/our_cell = get_cell()
-	if(!isnull(our_cell) && !(. & EMP_PROTECT_CONTENTS))
-		our_cell.emp_act(severity)
 
-/obj/item/inducer/attack_atom(obj/target, mob/living/carbon/user, params)
-	if(user.combat_mode)
-		return ..()
+/obj/item/inducer/Exited(atom/movable/gone, direction)
+	. = ..()
+	if(gone == powerdevice)
+		powerdevice = null
 
-	if(cantbeused(user))
-		return
+/obj/item/inducer/add_context(atom/source, list/context, obj/item/held_item, mob/user)
+	. = NONE
 
-	if(recharge(target, user))
+	if(isnull(held_item))
+		if(opened && !QDELETED(powerdevice))
+			context[SCREENTIP_CONTEXT_LMB] = "Remove Cell"
+			. = CONTEXTUAL_SCREENTIP_SET
 		return
 
-	return ..()
+	if(istype(held_item, /obj/item/stock_parts/power_store) && QDELETED(powerdevice))
+		context[SCREENTIP_CONTEXT_LMB] = "Insert cell"
+		return CONTEXTUAL_SCREENTIP_SET
 
-/obj/item/inducer/proc/cantbeused(mob/user)
-	if(!ISADVANCEDTOOLUSER(user))
-		to_chat(user, span_warning("You don't have the dexterity to use [src]!"))
-		return TRUE
+	if(istype(held_item, /obj/item/stack/sheet/mineral/plasma) && !QDELETED(powerdevice))
+		context[SCREENTIP_CONTEXT_LMB] = "Charge cell"
+		return CONTEXTUAL_SCREENTIP_SET
 
-	var/obj/item/stock_parts/power_store/our_cell = get_cell()
+	if(held_item.tool_behaviour == TOOL_SCREWDRIVER)
+		context[SCREENTIP_CONTEXT_LMB] = "[opened ? "Close" : "Open"] Panel"
+		return CONTEXTUAL_SCREENTIP_SET
 
-	if(isnull(our_cell))
-		balloon_alert(user, "no cell installed!")
-		return TRUE
+/obj/item/inducer/examine(mob/living/user)
+	. = ..()
 
-	if(!our_cell.charge)
-		balloon_alert(user, "no charge!")
-		return TRUE
-	return FALSE
+	if(!QDELETED(powerdevice))
+		. += span_notice("Its display shows: [display_energy(powerdevice.charge)].")
+		if(opened)
+			. += span_notice("The cell can be removed with an empty hand.")
+	else
+		. += span_warning("Its missing a power cell.")
 
-/obj/item/inducer/screwdriver_act(mob/living/user, obj/item/tool)
-	. = TRUE
-	tool.play_tool_sound(src)
+	. += span_notice("Its battery compartment can be [EXAMINE_HINT("screwed")] [opened ? "closed" : "opened"].")
```

Comment:
"screwed opened" ain't right, and it tends to be "screwed shut" rather than "screwed closed"
```suggestion
	. += span_notice("Its battery compartment can be [EXAMINE_HINT("screwed")] [opened ? "shut" : "open"].")
```